src/edu/ucsb/cs56/projects/utilities/calculator/Calculator.java

```diff
@@ -12,159 +13,172 @@
+	displayingResult = false;
+	this.display = display;
+	this.resultDisplay = resultDisplay;
+	resultDisplay.append("Hello!");
+	refresh();
+	
+	operators = new ArrayList<Character>();
+	numbers = new ArrayList<Double>();
+	currString = "";
+    }
 
     /**
        Call this method with a String to have the calculator do some operation
        (i.e. appending a digit to the current number, or appending an operator
        to the expression)
-     */
-	public void append(String s){
-		if(s.equals("-") && !onRightSide && left.equals(""))
-			left = left + s;
-		else if(s.equals("-") && onRightSide
-			&& !operator.equals("") && right.equals(""))
-			right = right + s; 
-
-		else if(functions.get(s) != null){   // Checks if symbol is in 'functions' HashMap
-			if (left.equals("") || left.equals("-"))
-				return;
-			else if(operator.equals("") || right.equals("")){
-				operator = s;
-				onRightSide = true;
-				displayingResult = false;
-			}
-		}
-		else if (s.equals(".") && getCurrentSide().contains(".")) // Don't allow more than 1 decimal per side
-			return;
-		else if (displayingResult){
-			left = s;
-			displayingResult = false;
-		} 
-		else if(onRightSide)
-			right = right + s;
-		else
-			left = left + s;
-		refresh();
+    */
+    public void append(String s){
+	
+	if (displayingResult){
+	    clear();
+	    displayingResult = false;
 	}
+	
+	// If s is a decimal point inside of a number or is a number itself
+	if ((!(currString.equals("")) && s.equals(".")) || Character.isDigit(s.charAt(0)))
+	    currString += s;
+	
+	// If the string is not empty (meaning it's not following a symbol)
+	else if (!(currString.equals("")))
+	    {
+		switch (s) {
+		case "+":
+		case "-":
+		case "*":
+		case "/":
+		case "^":
+		    double val;
+		if ((currString.substring(0,1)).equals("√"))
+		    val = Math.sqrt(Double.parseDouble(currString.substring(1,currString.length())));
+		else
+		    val = Double.parseDouble(currString);
+		numbers.add(val);
+		operators.add(s.charAt(0));
+		currString = "";
+		break;
+		default:
+		}
+	    }
+	
+	// If the user is signifying a negative number
+	else if ((s.charAt(0) == '-') || (s.charAt(0) == '√'))
+	    currString += s;
+	
+	refresh();
+    }
 
     /**
        Refresh the display to update it to the current state of the expression
-     */
-	public void refresh(){
-		if(onRightSide)
-			display.append(left + " " + operator + " "
-				       + right + "|");
-		else
-			display.append(left + "| " + operator + " " + right);
+    */
+    public void refresh(){
+	
+	String result = "";
+	if (operators == null)
+	    display.append(currString);
+	else if (operators.size() == 0)
+	    display.append(currString);
+	else {
+	    for (int i = 0; i < operators.size(); i++){
+		if (i > 0)
+		    result += " ";
+		result += ("" + numbers.get(i) + " " + operators.get(i));
+	    }
+	    if (!(currString.equals("")))
+		result += Double.toString(Double.parseDouble(currString));
```

Comment:
This has an unintended consequence of making numbers show up with a ".0" appended to the end without the user entering those characters. Just a small visual bug.